clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -1614,12 +1639,15 @@ scheduler_tmpl_t<RecordType, ReaderType>::pick_next_input_as_previously(
         }
         if (inputs_[index].reader->get_instruction_ordinal() <
                 segment.start_instruction &&
+            segment.start_instruction > 1 &&
```

Comment:
Looks like we're trying to avoid returning STATUS_WAIT in the case where the ROI segment starts at the beginning (i.e. at 1) and we're at the very beginning of the input (where get_instruction_ordinal returns 0?). Is that right?

Reply:
Yes.  Added a comment.